### PR TITLE
fix(popup): delete EventListener

### PR DESCRIPTION
**Pull request by 0x0dr1y**
This MR aims to fix the popup mixin. Currently there is a bug with creating and removing the eventListeners. This leads to the error `this.triggerlEl is not a function`. This MR relies on the official uiv implementation

### CODE REVIEW

**Comment by mtorromeo** [comment]
Hi, thanks for the fix.

Do you mind updating the MR to just include the fix itself without changing all the `addEventListener`/`removeEventListener` calls to `this.on`/`this.off`?

**Comment by 0x0dr1y** [comment]
> Hi, thanks for the fix.
> 
> Do you mind updating the MR to just include the fix itself without changing all the `addEventListener`/`removeEventListener` calls to `this.on`/`this.off`?

lol.. somehow i managed to miss that just the missing `removeEventListener` is the issue.. I'll adjust the fix later that day by just adding the missing `removeEventListener` :+1:

File: src/mixins/popupMixin.js [hunk 1]
```diff
@@ -294,17 +294,17 @@ export default {
...
+        this.triggerEl.removeEventListener('blur', this.hide);
+        this.triggerEl.removeEventListener('mouseenter', this.show);
+        this.triggerEl.removeEventListener('mouseleave', this.hide);
+        this.triggerEl.removeEventListener('click', this.toggle);
+        this.triggerEl.removeEventListener('mouseenter', this.handleAuto);
+        this.triggerEl.removeEventListener('mouseleave', this.handleAuto);
+        this.triggerEl.removeEventListener('focus', this.handleAuto);
+        this.triggerEl.removeEventListener('blur', this.handleAuto);
       }
       window('click', this.windowClicked);
```
Comment (mtorromeo) [inline src/mixins/popupMixin.js]: I think in your previous version you had a valid fix here also: `window` -> `window.removeEventListener`

**Comment by 0x0dr1y** [comment]
@mtorromeo i removed the unnecessary helper functions and just added the missing `removeEventListener` calls + changed `beforeDestroy` to `beforeUnmount`

(still not sure how i missed that there is already the eventlistener creation :joy:

**Review by mtorromeo — CHANGES_REQUESTED** [review]

**Review by mtorromeo — APPROVED** [review]

**Pull request merged**